From 34bf664d81cf234c866a647355b89ed6ba5abc0d Mon Sep 17 00:00:00 2001
From: Arpit Agarwal <arp@apache.org>
Date: Mon, 8 Aug 2016 12:29:24 -0700
Subject: [PATCH 1725/2748] HADOOP-13457. Remove hardcoded absolute path for
 shell executable. (Chen Liang)

(cherry picked from commit 2dbe16c0244222607048872dda2fbdd76ef655c9)

Change-Id: Ied70748251b2a684c1ff5a7fcef3bba1727aabea
---
 .../main/java/org/apache/hadoop/util/Shell.java    |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/Shell.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/Shell.java
index bae2ce8..87320f9 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/Shell.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/Shell.java
@@ -306,7 +306,7 @@ public static String appendScriptExtension(String basename) {
     String absolutePath = script.getAbsolutePath();
     return WINDOWS ?
       new String[] {"cmd", "/c", absolutePath }
-      : new String[] {"/bin/bash", bashQuote(absolutePath) };
+      : new String[] {"bash", bashQuote(absolutePath) };
   }
 
   /** a Unix command to set permission */
-- 
1.7.9.5

